### PR TITLE
Handle deleted teacher accounts in workshop summary reports

### PR DESCRIPTION
**Pull request by bencodeorg**
[This Honeybadger error](https://app.honeybadger.io/projects/3240/faults/65434817#notice-trace) occurred in a case where a user deleted their account, but the [account purging process](https://github.com/code-dot-org/code-dot-org/blob/staging/lib/cdo/delete_accounts_helper.rb#L97) hasn't run to remove their user ID from their attendances (runs 28 days after account is deleted.)

## Testing story

Added unit tests to cover uses of this method with deleted accounts.

## Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/test/factories/pd_factories.rb [hunk 1]
```diff
@@ -452,11 +452,16 @@
...
       end
-      if evaluator.cdo_scholarship_recipient
+
+      scholarship_params = {
+        user: teacher,
+        course: Pd::Workshop::COURSE_KEY_MAP[evaluator.workshop.course],
+        application_year: evaluator.workshop.school_year
+      }
+
+      if evaluator.cdo_scholarship_recipient && Pd::ScholarshipInfo.find_by(scholarship_params).nil?
```
Comment (molly-moen) [inline dashboard/test/factories/pd_factories.rb:466]: can you add a comment as to why the second check is needed?

**Review by molly-moen — APPROVED** [review]

File: dashboard/test/factories/pd_factories.rb (hunk 1 above)
Comment (bencodeorg) [reply dashboard/test/factories/pd_factories.rb:466]: Done! Take a look and LMK if unclear.

Comment (molly-moen) [reply dashboard/test/factories/pd_factories.rb:466]: looks good, thanks!

File: dashboard/test/factories/pd_factories.rb [hunk 2]
```diff
@@ -452,11 +452,20 @@
...
+        user: teacher,
+        course: Pd::Workshop::COURSE_KEY_MAP[evaluator.workshop.course],
+        application_year: evaluator.workshop.school_year
+      }
+
+      # We have an after_create hook on the enrollment model (see :set_default_scholarship_info)
+      # that creates a ScholarshipInfo entry in certain cases (namely, if the enrollment is in a CSF workshop).
+      # Skip creating a new ScholarshipInfo entry if one has already been created
+      # when the enrollment is created above.
+      if evaluator.cdo_scholarship_recipient && Pd::ScholarshipInfo.find_by(scholarship_params).nil?
```
Comment (hacodeorg) [inline dashboard/test/factories/pd_factories.rb:466]: nit: would it be shorter to use https://apidock.com/rails/v4.0.2/ActiveRecord/Relation/find_or_create_by?

**Review by hacodeorg — APPROVED** [review]

File: dashboard/test/factories/pd_factories.rb (hunk 2 above)
Comment (bencodeorg) [reply dashboard/test/factories/pd_factories.rb:466]: Would I be able to use the `pd_scholarship_info` factory that way?

Comment (bencodeorg) [reply dashboard/test/factories/pd_factories.rb:466]: I guess in this case I'm not making much use of the factory anyway.

**Pull request merged**